test/cpp/qps/qps_worker.cc

```diff
@@ -280,6 +280,7 @@ QpsWorker::QpsWorker(int driver_port, int server_port,
   gpr_atm_rel_store(&done_, static_cast<gpr_atm>(0));
 
   std::unique_ptr<ServerBuilder> builder = CreateQpsServerBuilder();
+  builder->AddChannelArgument(GRPC_ARG_ALLOW_REUSEPORT, 0);
```

Comment:
FTR, note that this only sets GRPC_ARG_ALLOW_REUSEPORT=0 for the worker_service (the one that listens on the driver port). Not sure if that's sufficient for avoiding the port collision or not (more reasoning is probably needed), just wanted to point that out.

Also, when we select both `driver_port` and `server_port` with `grpc_pick_unused_port_or_die()` (see above), it isn't clear why we are also setting GRPC_ARG_ALLOW_REUSEPORT=0 - what is it trying to prevent?
under normal circumstances, the worker service is exposed on the driver_port (which now uses  GRPC_ARG_ALLOW_REUSEPORT=0) before any benchmarking services are started (because the command to start a benchmarking service is only received through an already-running worker service.
So is the point of GRPC_ARG_ALLOW_REUSEPORT=0 to prevent accidentally trying to start benchmarking service on the same port? (which is a fair strategy, but this intention should be clear from the PR).

Btw, it's pretty discouraging how complicated our benchmarking harness is and how many subtle technicalities need to be reasoned about before we can figure out what's the right solution for fixing json_run_localhost. :-(

On a similar note, what is the reason for GRPC_ARG_ALLOW_REUSEPORT=1 being the default? As far as I can remember, this setting has caused a lot of trouble in the past, especially when testing (e.g. C# explicitly disables the reuseport in all its tests to avoid crosstalk between unrelated tests).